src/base/request.ml

```diff
@@ -28,9 +28,10 @@ module Handler = struct
    fun stack0 label_stack req0 ->
     let rec go req = function
       | [] ->
-          failwith
-            ( "Unhandled request: "
-            ^ Core_kernel.String.concat ~sep:"\n" label_stack )
+          raise
```

Comment:
As above, this should probably return an `option`, and the caller should handle the error by constructing an appropriate error message instead. This also has the benefit of never constructing this string if it isn't needed.

Reply:
good point! will change